notebooks/1_intelligent_agents_fall_2021/index.ipynb

```diff
@@ -0,0 +1,225 @@
+   "metadata": {},
+   "source": [
+    "#### Rationality vs perfection\n",
+    "Keep in mind that rationality is distinct from omniscience. an omniscience agent knows the actual outcome of its actions but in reality, an agent only knows the expected outcome of its action.\n",
+    "#### Autonomy\n",
+    "a rational agent should be autonomous meaning it mustn't only rely on the prior knowledge of its designer and must learn to compensate for partial or incorrect prior knowledge. In other words, rational agents should learn from experience. for example, in the vacuum world our agent could start to learn when the rooms usually get dirty based on its experience."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Task environment (PEAS)\n",
+    "we have already talked about performance measure, task environment, actuators and sensors. we group all these under the heading of the <b>Task enviroment </b> and we abbreviate it as <b>PEAS</b>(<b>P</b>erformance measure, <b>E</b>nviroment, <b>A</b>ctuators, <b>S</b>ensors). When designing an agent our first step should be specifying the task enviroment.\n",
+    "#### Types of environment\n",
+    "<ul>\n",
+    "  <li><b>Fully observable or partially observable</b> (do the agent sensors give access to the complete state of the environment at each time?)</li>\n",
+    "  <li><b>Single agent or multiagent</b> (are there more than one agent in the enviroment?)</li>\n",
+    "  <li><b>Deterministic or stochastic</b> (is the next state completely determined by the current state and the executed action?)</li>\n",
+    "  <li><b>Episodic or sequential</b> (is the agent's experience divided into atomic \"episodes“ where the choice of action in each episode depends only on the episode itself?)</li>\n",
+    "  <li><b>Discrete or continuous</b> (are there a limited number of distinct, clearly defined states, percepts and actions?)</li>  \n",
+    "</ul>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### PEAS example\n",
+    "here are a few example of specifying PEAS for different agents.\n",
+    "\n",
+    "| Agent       | Performance Measure | Environment |  Actuator | Sensor |\n",
+    "| ----------- | ----------- | ----------- | ----------- | ----------- |\n",
+    "| Hospital Management System       | Patient’s health, Admission process, Payment | Hospital, Doctors, Patients |  Prescription, Diagnosis, Scan report | Symptoms, Patient’s response |\n",
+    "| Automated Car Drive       | Comfortable trip, Safety, Maximum Distance | Roads, Traffic, Vehicles |  Steering wheel, Accelerator, Brake, Mirror | Camera, GPS, Odometer |\n",
+    "| Subject Tutoring       | Maximize scores, Improvement is students | Classroom, Desk, Chair, Board, Staff, Students |  Smart displays, Corrections | Eyes, Ears, Notebooks |"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Type of agents\n",
```

Comment:
Try to expand this part. There are other types of agents that aren't in slides but you can cover them here.